src/lib.rs

```diff
@@ -4,4 +4,3 @@ pub mod gfa;
 pub mod mmap;
 pub mod optfields;
 pub mod parser;
-pub mod writer;
```

Comment:
Wasn't sure what the desired outcome here was. You had the TODO for impl display, but I wasn't sure if the goal was to replace the current writer module functions with functions that just rely on display so as to make things backwards compatible. Happy to modify to accomodate the implementation you'd like. Just let me know.